samples/client/petstore/typescript-fetch/builds/one-of/models/ObjectCombinedOneOf.ts

```diff
@@ -0,0 +1,51 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Enum test
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import {
+    PatternObject,
+    PatternObjectTwo,
+    PatternObjectFromJSONTyped,
+    PatternObjectToJSON,
+    PatternObjectTwoFromJSONTyped,
+    PatternObjectTwoToJSON,
+} from './';
+
+/**
+ * @type ObjectCombinedOneOf
+ * 
+ * @export
+ */
+export type ObjectCombinedOneOf = PatternObject | PatternObjectTwo;
+
+export function ObjectCombinedOneOfFromJSON(json: any): ObjectCombinedOneOf {
+    return ObjectCombinedOneOfFromJSONTyped(json, false);
+}
+
+export function ObjectCombinedOneOfFromJSONTyped(json: any, ignoreDiscriminator: boolean): ObjectCombinedOneOf {
+    if ((json === undefined) || (json === null)) {
+        return json;
+    }
+    return PatternObjectFromJSONTyped(json, true) || PatternObjectTwoFromJSONTyped(json, true);
+}
+
+export function ObjectCombinedOneOfToJSON(value?: ObjectCombinedOneOf | null): any {
+    if (value === undefined) {
+        return undefined;
+    }
+    if (value === null) {
+        return null;
+    }
+    return PatternObjectToJSON(value as PatternObject) ?? PatternObjectTwoToJSON(value as PatternObjectTwo);
```

Comment:
Finally 🎉 Got that trouble code generated!

Before it was `{ ...TypeOneFromJson(value), ...TypeTwoFromJson(value) }` which was actually an all of? 🤔

AFAIK, With these changes we now have "its was completely parsed as Type one" or "its not TypeOne and completely parsed as TypeTwo"

Reply:
> we now have "its was completely parsed as Type one" or "its not TypeOne and completely parsed as TypeTwo"

erhm, not exactly. `TypeOneFromJson` only returns `null` if input is null, otherwise it returns an object So the second parsing (`PatternObjectTwoToJSON`) will never happen